java-frontend/src/main/java/org/sonar/java/se/checks/SwitchCaseWithoutBreakCheck.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se.checks;
+
+import com.google.common.collect.Lists;
+import org.sonar.check.Rule;
+import org.sonar.java.se.CheckerContext;
+import org.sonar.java.se.ProgramState;
+import org.sonar.plugins.java.api.tree.BlockTree;
+import org.sonar.plugins.java.api.tree.CaseLabelTree;
+import org.sonar.plugins.java.api.tree.IfStatementTree;
+import org.sonar.plugins.java.api.tree.StatementTree;
+import org.sonar.plugins.java.api.tree.SwitchStatementTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TryStatementTree;
+
+/**
+ * Rule implementation which checks for switch-case fall-through.
+ */
+@Rule(key = "S128")
+public class SwitchCaseWithoutBreakCheck extends SECheck {
+
+  private static final String ISSUE_MESSAGE
+    = "End this switch case with an unconditional break, return or throw statement.";
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    PreStatementVisitor visitor = new PreStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  private static boolean isUnconditionalExit(Tree syntaxNode) {
+    if (syntaxNode.is(
```

Comment:
It could be simpler to use a switch on the kind of the syntaxNode:
* For the unconditional trees, you return `true`.
* For `BLOCK`, `IF_STATEMENT` and `TRY_STATEMENT`, you do the specific operations from L66.
* For all the others (default), you return `false`.